### PR TITLE
docs: Configuring headers sent to subgraphs

### PR DESCRIPTION
**Pull request by abernix**
This follows up on https://github.com/apollographql/router/pull/453 with a
pass at the documentation.

### CODE REVIEW

File: docs/source/configuration.mdx [hunk 1]
```diff
@@ -124,32 +135,177 @@ server:
...
+- **`headers_propagate`**
+
+  This allows _selectively_ adding headers which were client-originated to the requests made to _specific_ subgraphs.
+
+  They can be configured based on a `matching` pattern using [`regex`]:
+
+  ```yaml
+      - headers_propagate:
+          matching:
+            regex: .*
```
Comment (BrynCooke) [inline docs/source/configuration.mdx:244]: Let's mention that host, content-length and content-type will not be propagate by regex:
https://github.com/apollographql/router/pull/505

File: docs/source/configuration.mdx [hunk 2]
```diff
@@ -124,32 +135,177 @@ server:
...
+          named:
+            name: "foo"
+            default_value: "bar"
+            rename: "fii"
+          named:
+            name: "Authorization"
+
+- **`headers_remove`**
+
+  The `headers_remove` layer allows removing a header that was client-originated and not sending it to a specific subgraph.  In the same way as `headers_propagate` can match either exactly or based on a regular expression using [`regex`].
```
Comment (BrynCooke) [inline docs/source/configuration.mdx:267]: Ditto for remove, host, content-length and content-type are not removed via regex.

File: docs/source/configuration.mdx [hunk 3]
```diff
@@ -124,32 +135,177 @@ server:
...
+
+  ```yaml
+      # Do not send this service the "Cookie" header.
+      - headers_remove:
+          name: "Cookie"
+      - headers_remove:
+          matching:
+            # Remove headers using the legacy 'x-' prefix.
+            regex: ^x-.*$
```
Comment (BrynCooke) [inline docs/source/configuration.mdx:276]: Suggest putting regexes in quotes to prevent yaml reserved character issues.

**Review by BrynCooke — APPROVED** [review]

File: docs/source/configuration.mdx (hunk 1 above)
Comment (garypen) [reply docs/source/configuration.mdx:244]: Are we case insensitive in our processing?

Comment (Geal) [reply docs/source/configuration.mdx:244]: it's compared with the http crate's `HeaderName`, which should handle that for us

File: docs/source/configuration.mdx (hunk 2 above)
Comment (Geal) [reply docs/source/configuration.mdx:267]: are all headers automatically propagated? Or should `headers_remove` only be used after a `headers_propagate` ?

File: docs/source/configuration.mdx (hunk 3 above)
Comment (abernix) [reply docs/source/configuration.mdx:276]: @BrynCooke I'll probably take this for a follow-up.  A few things that went into my thinking:

1. This wasn't quoted in the example I copied from the PR. 
2. I checked what CircleCI does here [for `filters`](https://circleci.com/docs/2.0/configuration-reference/#jobfilters) and they also didn't quote in their docs. 
3.	  ```yaml
	  # this is valid Yaml and is escaped properly.
	  my_key: this"is a "thing?'
	  ``` 

      I _might_ even argue that adding quotes actually makes things more difficult by forcing it?

4. Spaces are actually Would that be possible with header names?  Header keys [shouldn't have](https://stackoverflow.com/questions/47687379/what-characters-are-allowed-in-http-header-values) (which says, `token` = `CHAR` - `CTL` - `separators`) any characters that need escaping?

File: docs/source/configuration.mdx (hunk 1 above)
Comment (abernix) [reply docs/source/configuration.mdx:244]: I'll link to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers#hop-by-hop_headers

File: docs/source/configuration.mdx (hunk 2 above)
Comment (abernix) [reply docs/source/configuration.mdx:271]: I added 7b2864f53803067f548770457b4bac16f0a53388

File: docs/source/configuration.mdx (hunk 1 above)
Comment (abernix) [reply docs/source/configuration.mdx:248]: I added

**Pull request merged**